src/tests/integration/test_generic_db_operations.py

```diff
@@ -770,27 +753,140 @@ def test_sql_query(ip_with_dynamic_db, cell, request, test_table_name_dict):
-SELECT 1 AS column1, 2 AS column2
-"""
-    )
-    ip_with_dynamic_db.run_cell(
-        """
-    %%sql --save second_cte --no-execute
-SELECT
-  sum(column1),
-  sum(column2) FILTER (column2 = 2)
-FROM first_cte
-"""
+        "%%sql --save subset --no-execute \n"
+        f"SELECT * FROM {test_table_name_dict['numbers']}"
     )
+
+    out = ip_with_dynamic_db.run_cell(cell)
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db",
+    [
+        "ip_with_postgreSQL",
+        "ip_with_mySQL",
+        "ip_with_mariaDB",
+        "ip_with_SQLite",
+        pytest.param(
+            "ip_with_duckDB_native",
+            marks=pytest.mark.xfail(reason="Not yet implemented"),
+        ),
+        "ip_with_duckDB",
+        "ip_with_Snowflake",
+        pytest.param(
+            "ip_with_MSSQL", marks=pytest.mark.xfail(reason="Not yet implemented")
+        ),
+        pytest.param(
+            "ip_with_oracle", marks=pytest.mark.xfail(reason="Not yet implemented")
+        ),
+    ],
+)
+def test_sql_error_suggests_using_cte(ip_with_dynamic_db, request):
+    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+
     out = ip_with_dynamic_db.run_cell(
         """
     %%sql
-SELECT * FROM second_cte"""
+S"""
     )
     assert isinstance(out.error_in_exec, UsageError)
+    assert out.error_in_exec.error_type == "RuntimeError"
     assert CTE_MSG in str(out.error_in_exec)
+
+
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db",
+    [
+        "ip_with_SQLite",
+        pytest.param(
+            "ip_with_duckDB_native",
+            marks=pytest.mark.xfail(
+                reason="We're currently running each command in a new cursor"
+            ),
+        ),
+        "ip_with_duckDB",
+        "ip_with_postgreSQL",
+    ],
+)
+def test_temp_table(ip_with_dynamic_db, request):
+    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+    out = ip_with_dynamic_db.run_cell(
+        """%%sql
+create temp table my_table as select 42;
+select * from my_table;
+"""
+    )
+
+    assert out.error_in_exec is None
+    assert list(out.result) == [(42,)]
+
+
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db",
+    [
+        "ip_with_postgreSQL",
+        "ip_with_mySQL",
+        "ip_with_mariaDB",
+        "ip_with_SQLite",
+        "ip_with_duckDB_native",
+        "ip_with_duckDB",
+        "ip_with_Snowflake",
+        "ip_with_MSSQL",
+        "ip_with_oracle",
+    ],
+)
+def test_results_sets_are_closed(ip_with_dynamic_db, request, test_table_name_dict):
+    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+
+    ip_with_dynamic_db.run_cell(
+        f"""%%sql
+CREATE TABLE my_numbers AS SELECT * FROM {test_table_name_dict['numbers']}
+        """
+    )
+
+    ip_with_dynamic_db.run_cell(
+        """%%sql
+SELECT * FROM my_numbers
+        """
+    ).result
+
+    ip_with_dynamic_db.run_cell(
+        """%%sql
+DROP TABLE my_numbers
+        """
+    )
```

Comment:
Question: How we verify results_sets are really close if we only run the below statement in the end?
```
    ip_with_dynamic_db.run_cell(
        """%%sql
DROP TABLE my_numbers
        """
    )
```

Reply:
you're right. this test isn't doing what's supposed to do. I fixed it: the new test opens two sqlite connections to the same database, then it leaves a result set open (by running a select statement) and then tries to drop the table, if the drop is unsuccessful (due to open results), it'll raise an error

Reply:
Thank you! Looks better